airflow-core/src/airflow/ui/src/queries/useLogs.tsx

```diff
@@ -217,8 +220,23 @@ export const useLogs = (
     },
   );
 
+  const truncatedData = useMemo(() => {
+    if (!data?.content || typeof limit !== "number" || limit <= 0) {
+      return data;
+    }
+
+    const streamingContent = parseStreamingLogContent(data);
+    const truncatedContent =
+      streamingContent.length > limit ? streamingContent.slice(-limit) : streamingContent;
+
```

Comment:
Add a comment explaining why this is done in the front-end. (Backend does not support yet pagination / limits on logs reading endpoint)